content/basics/basics.mdx

```diff
@@ -0,0 +1,327 @@
+`useState` hook, the MobX local store is observable.
+
+Here is an example where we add an animal to a zoo using a form:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+const AddAnimalView = observer(({ zoo }: { zoo: Zoo }) => {
+  const local = useLocalStore(() => ({
+    name: '',
+    species: '',
+  }))
+
+  const handleAdd = () => {
+    zoo.addAnimal(new Animal(local.name, local.species))
+    // clear input again after adding
+    local.name = ''
+    local.species = ''
+  }
+
+  const handleChangeName = (e: React.ChangeEvent<HTMLInputElement>) => {
+    local.name = e.target.value
+  }
+
+  const handleChangeSpecies = (e: React.ChangeEvent<HTMLInputElement>) => {
+    local.species = e.target.value
+  }
+
+  return (
+    <div>
+      <div>
+        Name:
+        <input type="text" value={local.name} onChange={handleChangeName} />
+      </div>
+      <div>
+        Species:
+        <input
+          type="text"
+          value={local.species}
+          onChange={handleChangeSpecies}
+        />
+      </div>
+      <button onClick={handleAdd}>Add animal</button>
+    </div>
+  )
+})
+```
+
+Here we define an object called `local` with two observable properties, `name`
+and `species`. We use the React [controlled
+components](https://reactjs.org/docs/forms.html#controlled-components)
+technique to allow the user to edit these properties. As you can see, this
+simply involves mutating the local state directly. We also have an `Add animal`
+button which uses the local state to modify the zoo using the `addAnimal`
+action. It then needs to reset the local state again to allow the user to
+enter another animal.
+
+XXX interaction with strict mode, see https://github.com/mobxjs/mobx-react-lite/issues/235
+if strict mode is the recommended story, what would we recommend instead of
+use useLocalStore?
+
+## Application organization
+
+In these examples we described how you can create observer React components
+that take observable instances as props, and this is a good pattern to follow.
+
+But where do you get these props from? Often they can be retrieved from
+_other_ observable objects. Let's consider `ZooView`:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+const ZooView = observer(({ zoo }: { zoo: Zoo }) => {
+  return (
+    <div>
+      <AnimalsView animals={zoo.animals} />
+      <AddAnimalView zoo={zoo} />
+    </div>
+  )
+})
+```
+
+Where does `AnimalsView` get its `animals` prop from? From the `zoo` in
+`ZooView`. Where does `AnimalView` get its `animal` prop? From the each animal
+in the `AnimalsView`. We also pass `zoo` into `AddAnimalView` directly.
+
+But where would we put the `zoo`? One simple approach is to make it a global in
+your application, and simply import it when you need it.
+
+Often this is sufficient. Consider a `Locale` MobX model that describes information
+about your site's Locale. You can simply maintain the current locale as
```

Comment:
Let's warn here that global variables might be problematic for integration testing, it would cause flakiness of such tests.

Reply:
I think this section went against my own philosophy of "one way to do it" for this document by offering a lot of competing options, so I cut out the discussion of Context for now. I eventually want to add a section called "Context and cross-cutting concerns" but that can wait.

Reply:
Yea, context is certainly more advanced, and in simple cases passing through props works justice fine. But I would like to have here a clear notion that stores in global variables might bite you.